Nightlies Apple Clang: Error SONARScoring

```
KNIME_PACKAGING_TRUNK/OpenMS/src/openms/source/ANALYSIS/OPENSWATH/SONARScoring.cpp:279:11: error: use of undeclared identifier 'isnan'; did you mean 'std::isnan'?
      if (isnan(rsq)) rsq = 0.0; // check for nan
          ^~~~~
          std::isnan
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk/usr/include/c++/4.2.1/cmath:551:5: note: 'std::isnan' declared here
    isnan(_Tp __f) { return ::__gnu_cxx::__capture_isnan(__f); }
    ^
1 error generated.
```

@hroest can you have a look at this?

--- a/src/openms/source/ANALYSIS/OPENSWATH/SONARScoring.cpp
+++ b/src/openms/source/ANALYSIS/OPENSWATH/SONARScoring.cpp
@@ -42,6 +42,7 @@
 
 #include <OpenMS/ANALYSIS/OPENSWATH/OPENSWATHALGO/ALGO/Scoring.h>
 #include <cmath>
+#include <boost/math/special_functions/fpclassify.hpp> // for isnan
 
 // #define DEBUG_SONAR
 
@@ -276,7 +277,7 @@ namespace OpenMS
       std::vector<double> xvals;
       for (Size pr_idx = 0; pr_idx < sonar_profile_pos.size(); pr_idx++) {xvals.push_back(pr_idx);}
       double rsq = OpenSwath::cor_pearson( xvals.begin(), xvals.end(), sonar_profile_pos.begin() );
-      if (isnan(rsq)) rsq = 0.0; // check for nan
+      if (boost::math::isnan(rsq)) rsq = 0.0; // check for nan
 
       // try to find largest diff
       double sonar_largediff = 0.0;

